physics/dynamic_frame_body.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "physics/dynamic_frame.hpp"
+
+namespace principia {
+namespace physics {
+template <typename FromFrame, typename ToFrame>
+RigidMotion<FromFrame, ToFrame>::RigidMotion(
+    RigidTransformation<FromFrame, ToFrame> const& rigid_transformation,
+    AngularVelocity<ToFrame> const& rotation,
+    Velocity<ToFrame> const& translation)
+    : rigid_transformation_(rigid_transformation),
+      rotation_(rotation),
+      translation_(translation) {}
+
+template <typename FromFrame, typename ToFrame>
+RigidTransformation<FromFrame, ToFrame> const&
+RigidMotion<FromFrame, ToFrame>::rigid_transformation() const {
+  return rigid_transformation_;
+}
+
+template <typename FromFrame, typename ToFrame>
+OrthogonalMap<FromFrame, ToFrame> const&
+RigidMotion<FromFrame, ToFrame>::orthogonal_map() const {
+  return rigid_transformation_.linear_map();
+}
+
+template <typename FromFrame, typename ToFrame>
+DegreesOfFreedom<ToFrame> RigidMotion<FromFrame, ToFrame>::operator()(
+    DegreesOfFreedom<FromFrame> const& degrees_of_freedom) const {
+  Position<ToFrame> const result_position =
```

Comment:
Is this intermediate variable needed?  I would just put the evaluation at line 33.